### PR TITLE
Adding modal for the marker points

### PR DESCRIPTION
**Pull request by harikautilya**
# Fill out the following while publishing a new pull request.

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Added BottomSheet modal for the UI  . 

* **What is the current behavior?** (You can also link to an open issue here)
https://github.com/UdacityAndroidDevScholarship/blood-donation/issues/40

* **What is the new behavior (if this is a feature change)?**
When the user clicks on any of the marker in the map, it opens a new layout at the bottom of the map.
The layout can be populated depending on the marker type.
Please check if are okie with this design , then i shall go ahead with design the donor and receiver UI for the bottomsheet population.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/HomeActivity.java [hunk 1]
```diff
@@ -22,20 +27,47 @@
 /**
  * Created by Ankush Grover(ankushgrover02@gmail.com) on 23/04/2018.
  */
-public class HomeActivity extends BaseActivity implements OnMapReadyCallback, HomeContract.View {
+public class HomeActivity extends BaseActivity implements OnMapReadyCallback, HomeContract.View, GoogleMap.OnMarkerClickListener {
 
     private GoogleMap mMap;
     private HomePresenter presenter;
+    private LinearLayout bottomSheet;
```
Comment (nipunjindal) [inline app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/HomeActivity.java]: member variables should follow `m<Variable>` convention.

File: app/src/main/res/layout/activity_home.xml [hunk 2]
```diff
@@ -9,53 +9,25 @@
...
         android:layout_width="match_parent"
         android:layout_height="match_parent"
         tools:context=".ui.home.HomeActivity">
 
+        <android.support.design.widget.CoordinatorLayout xmlns:android="http://schemas.android.com/apk/res/android"
+            xmlns:app="http://schemas.android.com/apk/res-auto"
+            xmlns:tools="http://schemas.android.com/tools"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:background="#efefef"
```
Comment (nipunjindal) [inline app/src/main/res/layout/activity_home.xml]: move to constant

Comment (harikautilya) [reply app/src/main/res/layout/activity_home.xml]: Done

**Comment by harikautilya** [comment]
Finished all the reviews mentioned...  @nipunjindal

**Comment by harikautilya** [comment]
Hey will updating more of this PR

**Comment by Bloody-Badboy** [comment]
@harikautilya fix merge conflicts

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -20,18 +20,22 @@
...
+
+            <category android:name="android.intent.category.LAUNCHER"/>
+        </intent-filter>
     </activity>
 
     <activity
         android:name="com.udacity.nanodegree.blooddonation.ui.splash.SplashScreenActivity"
         android:screenOrientation="portrait"
         android:theme="@style/AppTheme.NoActionBar">
-      <intent-filter>
```
Comment (ritesh-singh) [inline app/src/main/AndroidManifest.xml]: This should be the launcher right?

Comment (ritesh-singh) [inline app/src/main/AndroidManifest.xml]: Why this change? This should be the launcher right?

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -20,18 +20,22 @@
     <activity
         android:name="com.udacity.nanodegree.blooddonation.ui.home.view.HomeActivity"
         android:screenOrientation="portrait">
+        <intent-filter>
```
Comment (ritesh-singh) [inline app/src/main/AndroidManifest.xml]: Remove this intent filter.

File: app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/view/HomeActivity.java [hunk 5]
```diff
@@ -15,98 +18,179 @@
...
+    private BottomSheetBehavior<LinearLayout> donorBehavior;
+    private BottomSheetBehavior<LinearLayout> receiverBehaviour;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_home);
+
+        mPresenter = new HomeActivityPresenter(this);
+        mDonorSheet = findViewById(R.id.donor_sheet);
```
Comment (ritesh-singh) [inline app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/view/HomeActivity.java]: You can use data binding here, instead of doing findViewById.

File: app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/view/HomeActivity.java [hunk 6]
```diff
@@ -15,98 +18,179 @@
...
 
 /**
  * Created by Ankush Grover(ankushgrover02@gmail.com) on 23/04/2018.
  */
-public class HomeActivity extends BaseActivity
-    implements OnMapReadyCallback, HomeActivityContract.View {
+public class HomeActivity extends BaseActivity implements OnMapReadyCallback, HomeActivityContract.View, GoogleMap.OnMarkerClickListener {
+
+    private GoogleMap mMap;
+    private HomeActivityPresenter mPresenter;
```
Comment (ritesh-singh) [inline app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/view/HomeActivity.java]: Code to interface, not implementation, use HomeContract.Presenter.

File: app/src/main/res/layout/activity_home.xml [hunk 7]
```diff
@@ -1,67 +1,36 @@
...
 
-  <data>
+    <data>
 
-    <variable
-        name="presenter"
-        type="com.udacity.nanodegree.blooddonation.ui.home.HomeActivityContract.Presenter"
-        />
-  </data>
+        <variable
```
Comment (ritesh-singh) [inline app/src/main/res/layout/activity_home.xml]: Again, HomeContract Presenter here.

File: app/src/main/res/values/api_keys.xml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (ritesh-singh) [inline app/src/main/res/values/api_keys.xml]: Delete this file, take update from develop.
Create app/src/debug/res/values/google_maps_api.xml and add key here.

**Comment by ritesh-singh** [comment]
@harikautilya Please, do the required changes. I have proceed with geofire.

File: app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/view/HomeActivity.java (hunk 5 above)
Comment (harikautilya) [reply app/src/main/java/com/udacity/nanodegree/blooddonation/ui/home/view/HomeActivity.java]: No over here it is not possible

**Comment by harikautilya** [comment]
@ritesh-singh  made all changes requested

File: app/src/main/res/layout/activity_home.xml (hunk 7 above)
Comment (harikautilya) [reply app/src/main/res/layout/activity_home.xml]: it is Home contract presenter ....do u want me to rename it?

**Comment by ritesh-singh** [comment]
Cool, i made and pushed. Please, take update.
